hax/hax/handler.py

```diff
@@ -61,23 +61,70 @@ def stop(self) -> None:
         self.is_stopped = True
 
     @repeat_if_fails(wait_seconds=1)
-    def _update_process_status(self, event: ConfHaProcess) -> None:
+    def _update_process_status(self, q: Queue, event: ConfHaProcess) -> None:
         # If a consul-related exception appears, it will
         # be processed by repeat_if_fails.
         #
         # This thread will become blocked until that
         # intermittent error gets resolved.
         self.consul.update_process_status(event)
+        svc_status = m0HaProcessEvent.event_to_svchealth(event.chp_event)
+        if event.chp_type == m0HaProcessType.M0_CONF_HA_PROCESS_M0D:
+            # Broadcast the received motr process status to other motr
+            # processes in the cluster.
+            q.put(BroadcastHAStates(states=[HAState(fid=event.fid,
+                                    status=svc_status)],
+                  reply_to=None))
 
-    def update_process_failure(self, ha_states: List[HAState]) -> None:
+    def update_process_failure(self, q: Queue,
```

Comment:
It seems like the method needs to be decorated with `@repeat_if_fails()`
Reason: the code within can raise `HAConsistencyException`. The method is invoked directly from `qconsumer` thread (handler.py) and it seems like the exception is not handled there. So `qconsumer` will pull the message from the thread Queue and once the exception raises, the message will be forgotten and will not be processed afterwards.